src/App.js

```diff
@@ -1,16 +1,31 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [likedMessages, setLikedMessages] = useState(0);
+
+  const likeMessage = () => {
+    setLikedMessages(likedMessages => likedMessages + 1);
+  };
+
+  const unlikeMessage = () => {
+    setLikedMessages(likedMessages => likedMessages - 1);
+  };
+  
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>Chat Messages</h1>
+        <h2>{likedMessages} ❤️s</h2>
       </header>
       <main>
-        {/* Wave 01: Render one ChatEntry component
-        Wave 02: Render ChatLog component */}
+        <ChatLog
+          entries={chatMessages}
```

Comment:
Instead of just passing in chatMessages from the messages.json file, we should use state as a single source of truth for keeping track of the data related to messages. Therefore, we should have a piece of state on line 8 like: 
```javascript
const [chatMessageData, setChatMessagData] = useState(chatMessages);
```

You'd also need an event handler that you can pass down to ChatEntry.js that would help you toggle the liked value on a single chat. It would be better for the logic about how to copy the message and change the liked status to be written in App.js

```javascript
const updateEntryData = id => {
const entries = chatMessageData.map(entry => {
      if (entry.id === id){
        return {...entry, liked: !entry.liked};
      } 
      else {
        return entry;
      }
    });
    setChatMessagData(entries);
}